api/desecapi/authentication.py

```diff
@@ -48,6 +48,11 @@ def authenticate_credentials(self, basic):
         try:
             user, key = base64.b64decode(basic).decode(HTTP_HEADER_ENCODING).split(':')
             token = self.model.objects.get(key=key)
+            if '@' in user:
```

Comment:
We could make the check here more restrictive and more explicit by checking that it is one of a list of allowed values. Allowed values could be: any of the user's domain names, empty string, the user's email address.

    allowed_values = ['', token.user.email] + [domain.name for domain in token.user.domains.all()]
    if user not in allowed_values:

Reply:
neat! For domain names, we need to consider the lowercase username, so I made small adaption.